app/src/main/java/de/westnordost/streetcomplete/osm/Surface.kt

```diff
@@ -1,12 +1,30 @@
 package de.westnordost.streetcomplete.osm
 
-val ANYTHING_UNPAVED = setOf(
+val SOFT_SURFACES = setOf("ground", "earth", "dirt", "grass", "sand", "mud", "ice", "salt", "snow", "woodchips")
+
+val ANYTHING_UNPAVED = SOFT_SURFACES + setOf(
     "unpaved", "compacted", "gravel", "fine_gravel", "pebblestone", "grass_paver",
-    "ground", "earth", "dirt", "grass", "sand", "mud", "ice", "salt", "snow", "woodchips"
 )
 
-val ANYTHING_PAVED = setOf(
+val ANYTHING_FULLY_PAVED = setOf(
     "paved", "asphalt", "cobblestone", "cobblestone:flattened", "sett",
-    "concrete", "concrete:lanes", "concrete:plates", "paving_stones",
+    "concrete", "concrete:plates", "paving_stones",
     "metal", "wood", "unhewn_cobblestone"
 )
+
+val ANYTHING_PAVED = ANYTHING_FULLY_PAVED + setOf(
+    "concrete:lanes"
+)
+
+fun isSurfaceAndTracktypeMismatching(surface: String, tracktype: String): Boolean {
+    if (tracktype == "grade1" && ANYTHING_UNPAVED.contains(surface)) {
+        return true
+    }
+    if (tracktype == "grade2" && SOFT_SURFACES.contains(surface)) {
+        return true
+    }
+    if ((tracktype == "grade3" || tracktype == "grade4" || tracktype == "grade5") && ANYTHING_FULLY_PAVED.contains(surface)) {
```

Comment:
Not sure if this really improves readability, but just to get out the suggestion:

```kt
private fun String.tracktypeGrade(): Int = this.replaceFirst("grade", "").asIntOrNull() ?: -1

// then later:
if (tracktype.tracktypeGrade() in 3..5 && surface in ANYTHING_FULLY_PAVED) {
```

or maybe:

```kt
if (tracktype in setOf("grade3", "grade4", "grade5") && surface in ANYTHING_FULLY_PAVED) {
```

Reply:
but this creates a `HashSet` every time it is called